drivers/msgbox/sunxi-msgbox.c

```diff
@@ -46,67 +54,41 @@ set_handler(struct device *dev, uint8_t chan, msgbox_handler handler)
 	((msgbox_handler *)dev->drvdata)[chan] = handler;
 }
 
-static void
-sunxi_msgbox_handle_msg(struct device *dev, uint8_t chan)
+static bool
+sunxi_msgbox_peek_data(struct device *dev, uint8_t chan)
 {
-	msgbox_handler handler;
-	uint32_t msg = mmio_read32(dev->regs + RECV_MSG_DATA_REG(chan));
+	uint32_t reg;
 
-	if ((handler = get_handler(dev, chan)))
-		handler(dev, chan, msg);
-	else
-		debug("%s: Unsolicited message %08x in channel %d",
-		      dev->name, msg, chan);
-}
-
-static void
-sunxi_msgbox_irq(struct device *dev)
-{
-	uint32_t reg = mmio_read32(dev->regs + IRQ_STATUS_REG);
+	assert(chan < SUNXI_MSGBOX_CHANS);
```

Comment:
Why assert the channel rather than using a control structure to verify the channel is within the bounds before reading? It seems like this would not only be safer but more descriptive (especially when including debug statements).

Reply:
Because the channel is not variable at runtime (i.e. SCPI is always going to use channel 0 for Linux and channel 1 for ATF). The asserts are here to catch programmer error.

Reply:
Thanks for the explanation. I will make sure this is covered in `contributing.md` as a _standard_ of this project.